hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -655,3 +736,19 @@ func versionOf(url *url.URL) (string, error) {
 	// No version found, that's fine
 	return "", nil
 }
+
+func appendIfUniqueType(slice []astmodel.Type, item astmodel.Type) []astmodel.Type {
```

Comment:
There's a standard Go idiom where you could use `map[Type]bool` as a set; would that make things easier/better?

Reply:
It’s actually `map[Type]struct{}`; my version isn’t as simple 😆

Reply:
The problem is we want the results to be strictly ordered according to the order they were defined in the oneOf (to aid with deterministic code generation). They also need to be indexed because we sometimes use the index to help generate a unique property name. I think the combination of these two things means that using a map is a bit awkward as we lose our ordering guarantee.

Reply:
Ok, fair enough.

Reply:
It would be nice if maps worked like Python dicts (by default, they iterate through in insertion order) - unfortunately though they don't behave like that